flink-formats/flink-orc/src/test/java/org/apache/flink/orc/OrcFileSystemFilterTest.java

```diff
@@ -74,5 +75,20 @@ public void testApplyPredicate() {
         OrcFilters.Predicate predicate6 =
                 new OrcFilters.LessThan("long1", PredicateLeaf.Type.LONG, 10);
         assertThat(predicate5).hasToString(predicate6.toString());
+
+        // and
+        CallExpression andExpression =
+                CallExpression.permanent(
+                        BuiltInFunctionDefinitions.AND,
+                        Arrays.asList(greaterExpression, lessExpression),
+                        DataTypes.BOOLEAN());
+        OrcFilters.Predicate predicate7 = OrcFilters.toOrcPredicate(andExpression);
+        OrcFilters.Predicate predicate8 =
+                new OrcFilters.And(
+                        new OrcFilters.Not(
+                                new OrcFilters.LessThanEquals(
+                                        "long1", PredicateLeaf.Type.LONG, 10)),
+                        new OrcFilters.LessThan("long1", PredicateLeaf.Type.LONG, 10));
```

Comment:
Does `new OrcFilters.And(predicate4, predicate6);` also work? If so, it would be more clean, WDYT?